string behavior
string arm
float32[] min_errors
float32[] max_errors
